### PR TITLE
Documentation and Init_restart() update

### PR DESCRIPTION
**Pull request by BvB93**
Documentation
---------------
*  Added requirements.txt with Sphinx>=2.1; this should fix the Sphinx version issue on [readthedocs.org](https://readthedocs.org/projects/qmflows/builds/10549955/).

``init_restart()``
----------------
* Introduced a couple of code-style improvements.
* Added the ``load_jobs`` keyword which automatically loads all previous jobs into the jobmanager.
* Fix: Ensure that that ``config.default_jobmanager.names`` is properly set.

### CODE REVIEW

File: src/qmflows/utils.py [hunk 1]
```diff
@@ -53,31 +55,52 @@ def dict2Setting(d):
     return r
 
 
-def init_restart(path: Union[None, AnyStr, os.PathLike] = None,
-                 folder: Union[None, AnyStr, os.PathLike] = None) -> None:
+def init_restart(path: Union[None, str, os.PathLike] = None,
+                 folder: Union[None, str, os.PathLike] = None,
+                 load_jobs: bool = False) -> None:
```
Comment (BvB93) [inline src/qmflows/utils.py]: Do you feel the default value of this agument should be ``True``?
Automatically loading previous jobs is nice, but the pickling process can be quite slow.

Comment (felipeZ) [reply src/qmflows/utils.py]: It depends on the size of the jobs that are most commonly run. It the most common jobs are significantly large then I think we should make it `False` by default

File: src/qmflows/utils.py [hunk 2]
```diff
@@ -53,31 +55,52 @@ def dict2Setting(d):
...
     # Parse variables
-    path_ = os.getcwd() if path is None else os.path.abspath(path)
-    folder_ = 'plams_workdir' if folder is None else os.path.normpath(folder)
-    workdir = os.path.join(path_, folder_)
+    path_ = os.getcwd() if path is None else abspath(path)
+    folder_ = 'plams_workdir' if folder is None else normpath(folder)
+    workdir = join(path_, folder_)
+
+    # The JobManager instance
+    jobmanager = config.default_jobmanager
```
Comment (felipeZ) [inline src/qmflows/utils.py]: this variable makes the code easier to read

File: src/qmflows/utils.py (hunk 1 above)
Comment (BvB93) [reply src/qmflows/utils.py]: Indeed, I feel it's definitely the safer option of the two.

**Pull request merged**